lib/Mir.ml

```diff
@@ -13,76 +13,59 @@ open Core_kernel
 
 type litType = Int | Real | Str [@@deriving sexp, hash]
 
```

Comment:
I am not sure it is a good idea to fully remove the operator type. In particular, `Or` and `And` do not behave like functions semantically, as they are lazy in their second argument. A lot of optimizations need to special case on them for that reason to make sure the optimizations are semantics preserving.

Reply:
We talked about this in the meeting though - we can roughly as easily pattern match on `| FunApp("Or", lhs :: rhs :: [])` as `BinOp(lhs, Or, rhs)`. I assume the optimizer is keeping track of which functions are commutative and generally not assuming that order doesn't matter right?

Reply:
Sure, you can also pattern match on those. You just need to remember to do so, because they will usually need special treatment.
It's also a bit slower as it will require string comparison, but that presumably doesn't matter.